srml/staking/src/phragmen.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Rust implementation of the Phragmén election algorithm.
+
+use rstd::{prelude::*};
+use primitives::Perquintill;
+use primitives::traits::{Zero, As};
+use parity_codec::{HasCompact, Encode, Decode};
+use crate::{Exposure, BalanceOf, Trait, ValidatorPrefs, IndividualExposure};
+
+// Wrapper around validation candidates some metadata.
+#[derive(Clone, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Candidate<AccountId, Balance: HasCompact> {
+	// The validator's account
+	pub who: AccountId,
+	// Exposure struct, holding info about the value that the validator has in stake.
+	pub exposure: Exposure<AccountId, Balance>,
+	// Accumulator of the stake of this candidate based on received votes.
+	approval_stake: Balance,
+	// Intermediary value used to sort candidates.
+	// See Phragmén reference implementation.
+	pub score: Perquintill,
+}
+
+// Wrapper around the nomination info of a single nominator for a group of validators.
+#[derive(Clone, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Nominations<AccountId, Balance: HasCompact> {
+	// The nominator's account.
+	who: AccountId,
+	// List of validators proposed by this nominator.
+	nominees: Vec<Vote<AccountId, Balance>>,
+	// the stake amount proposed by the nominator as a part of the vote.
+	// Same as `nom.budget` in Phragmén reference.
+	stake: Balance,
+	// Incremented each time a nominee that this nominator voted for has been elected.
+	load: Perquintill,
+}
+
+// Wrapper around a nominator vote and the load of that vote.
+// Referred to as 'edge' in the Phragmén reference implementation.
```

Comment:
```suggestion
//
// Referred to as 'edge' in the Phragmén reference implementation.
```

You need to have an empty line for line breaks.